Src/Particle/AMReX_Particle.H

```diff
@@ -33,6 +33,79 @@ namespace amrex {
 
     using namespace ParticleIdCpus;
 
+    namespace particle_impl {
+
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    Long unpack_id (const uint64_t idcpu) noexcept {
```

Comment:
Any reasons to only use this new function in `ParticleIDWrapper`, but not in `ConstParticleIDWrapper`?